test/functional/tests/fault_injection/test_fault_injection_with_mounted_core.py

```diff
@@ -142,27 +137,61 @@ def test_add_occupied_core(cache_mode):
     with TestRun.step("Prepare two caches and one core device."):
         cache_dev = TestRun.disks['cache']
         cache_dev.create_partitions([Size(2, Unit.GibiByte), Size(2, Unit.GibiByte)])
-        cache_dev1 = cache_dev.partitions[0]
-        cache_dev2 = cache_dev.partitions[1]
+        cache_part1 = cache_dev.partitions[0]
+        cache_part2 = cache_dev.partitions[1]
         core_dev = TestRun.disks['core']
         core_dev.create_partitions([Size(4, Unit.GibiByte)])
-        core_dev = core_dev.partitions[0]
+        core_part = core_dev.partitions[0]
 
-    with TestRun.step("Start first cache instance"):
-        cache1 = casadm.start_cache(cache_dev1, cache_mode, force=True)
+    with TestRun.step("Start the first cache instance"):
+        cache1 = casadm.start_cache(cache_part1, cache_mode, force=True)
 
     with TestRun.step("Add core device to first cache instance."):
-        core = cache1.add_core(core_dev)
+        core = cache1.add_core(core_part)
+
+    with TestRun.step("Start the second cache instance"):
+        cache2 = casadm.start_cache(cache_part2, cache_mode, force=True)
+
+    with TestRun.step("Try adding the same core device to the second cache instance."):
+        output = TestRun.executor.run_expect_fail(
+            cli.add_core_cmd(cache_id=str(cache2.cache_id), core_dev=str(core_part.system_path),
+                             core_id=str(core.core_id)))
+        cli_messages.check_msg(output, cli_messages.add_cached_core)
+
+    with TestRun.step("Stop caches."):
+        casadm.stop_all_caches()
+
+
+@pytest.mark.parametrize("cache_mode", CacheMode)
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_add_same_core_twice(cache_mode):
```

Comment:
I noticed this test could be merged with test_add_occupied_core, just by using two steps in one test - try to add to the same cache and try to add to another cache